packages/trader/src/Modules/Trading/Containers/trade-params-mobile.jsx

```diff
@@ -20,135 +20,133 @@ const DEFAULT_DURATION = Object.freeze({
 const makeGetDefaultDuration = (trade_duration, trade_duration_unit) => duration_unit =>
     trade_duration_unit === duration_unit ? trade_duration : DEFAULT_DURATION[duration_unit];
 
-class TradeParamsModal extends React.Component {
-    constructor(props) {
-        super(props);
-        const { amount, duration, duration_unit } = this.props;
-        const getDefaultDuration = makeGetDefaultDuration(duration, duration_unit);
+const TradeParamsModal = props => {
+    const { amount, duration, duration_unit } = props;
 
-        this.state = {
-            trade_param_tab_idx: 0,
-            duration_tab_idx: undefined,
-            amount_tab_idx: undefined,
-            has_amount_error: false,
-            has_duration_error: false,
-            // duration unit values
-            curr_duration_unit: duration_unit,
-            curr_duration_value: duration,
-            t_duration: getDefaultDuration('t'),
-            s_duration: getDefaultDuration('s'),
-            m_duration: getDefaultDuration('m'),
-            h_duration: getDefaultDuration('h'),
-            d_duration: getDefaultDuration('d'),
-            // amount values
-            stake_value: amount,
-            payout_value: amount,
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const getDefaultDuration = React.useCallback(makeGetDefaultDuration(duration, duration_unit), []);
+
+    const [state, setState] = React.useState({
+        trade_param_tab_idx: 0,
+        duration_tab_idx: undefined,
+        amount_tab_idx: undefined,
+        has_amount_error: false,
+        has_duration_error: false,
+        // duration unit values
+        curr_duration_unit: duration_unit,
+        curr_duration_value: duration,
+        t_duration: getDefaultDuration('t'),
+        s_duration: getDefaultDuration('s'),
+        m_duration: getDefaultDuration('m'),
+        h_duration: getDefaultDuration('h'),
+        d_duration: getDefaultDuration('d'),
+        // amount values
+        stake_value: amount,
+        payout_value: amount,
+    });
```

Comment:
I think this is a good case of using `useReducer` over this. [have a look](https://reactjs.org/docs/hooks-reference.html#usereducer)

Reply:
Modified this to `useReducer`, but it updates the state directly with payload. I don't want to complicate it by using actions

Reply:
IMO I don't think `useReducer` should be used here (the state values are simple primitives) but instead all of the state entries should be refactored to use their own `useState` hook, i.e. ✂️🌽

Reply:
I don't want to have `useState` hook for each entry because we can update multiple duration state with a single line.
```
dispatch({
    [`${selected_duration_unit}_duration`]: selected_duration
})
```
Plus, there is no performance overhead with this approach